format/spec.md

```diff
@@ -1680,6 +1680,23 @@ Row-level delete changes:
     * These position delete files must be merged into the DV for a data file when one is created
     * Position delete files that contain deletes for more than one data file need to be kept in table metadata until all deletes are replaced by DVs
 
+Row lineage changes:
+
+* Writers must set the table's `next-row-id` and use the existing `next-row-id` as the `first-row-id` to create a new snapshot
+    * When a table is upgraded to v3, `next_row_id` should be initialized to 0
+    * It is recommended to increment `next-row-id` by the total `added_rows_count` and `existing_rows_count` of all manifests assigned a `first_row_id`
```

Comment:
Maybe add a line
* When committing a new snapshot `next-row-id` must be incremented by at least the number of newly assigned `row-id` in the snapshot

Reply:
Added. And I copied this for manifest lists as well ("at least the number of newly assigned row ids in each manifest")